src/commands/dev/index.ts

```diff
@@ -493,15 +493,19 @@ export default class Dev extends BaseCommand {
 
   private async getAvailablePort(port: number): Promise<number> {
     while (!(await PortUtil.isPortAvailable(port))) {
+      let message = `Trying to listen on port ${port}, but something is already using it. What port would you like us to run the API gateway on (you can use the '--port' flag to skip this message in the future)?`;
+      if (RESTRICTED_PORTS.has(port)) {
+        message = `Port ${port} is restricted by browsers. What port would you like us to run the API gateway on instead?`;
+      }
       const answers = await inquirer.prompt([
         {
           type: 'input',
           name: 'port',
-          message: `Trying to listen on port ${port}, but something is already using it. What port would you like us to run the API gateway on (you can use the '--port' flag to skip this message in the future)?`,
+          message,
           validate: (value) => {
             const port = Number.parseInt(value);
-            if (!this.isValidPort(port)) {
-              return 'Port must be 80, 443, or any port between 1024 and 66535.';
+            if (!port || port <= 0 || port > 65535) {
```

Comment:
We also need the restricted port check here I believe.

Reply:
We do this question in a loop https://github.com/architect-team/architect-cli/pull/908/files#diff-c803da602bbb1783595bc04de6a99e122c48269d8c8eb1c6df6397bf5360f91bR495 so technically it still works if you type a restricted port (and asks the question again), though better UX to specifically say "Yo this port is restricted" - will update that 👍